Assets/00_Content/Scripts/Vikings/States/SatisfiedVikingState.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Linq;
+using UnityEngine;
+using Utilities;
+
+namespace Vikings.States {
+	public class SatisfiedVikingState : VikingState {
+		private const float DropDelay = 0.15f;
+
+		private float satisfiedDuration;
+		private float satisfiedTimer;
+
+		private int coinsToDrop;
+		private float dropTimer;
+
+		private bool IsDroppingCoins => coinsToDrop > 0;
+
+		public SatisfiedVikingState(Viking viking) : base(viking) { }
+
+		public override VikingState Enter() {
+			satisfiedDuration =
+				Random.Range(viking.Data.satisfiedDurationMinMax.x, viking.Data.satisfiedDurationMinMax.y);
+			satisfiedTimer = satisfiedDuration;
+
+			return this;
+		}
+
+		public override VikingState Update() {
+			VikingState nextState = this;
+			satisfiedTimer -= Time.deltaTime;
+
+			if (satisfiedTimer <= 0 && !IsDroppingCoins)
+				SetupDroppingCoins();
+
+			if (IsDroppingCoins)
+				nextState = DropCoins();
+
+			return nextState;
+		}
+
+		private void SetupDroppingCoins() {
+			if (viking.CurrentDesireIndex < viking.Desires.Length) {
+				coinsToDrop = CalculateCoinsToDrop(viking.Stats.Mood);
+			}
+			else {
+				float avgMood = viking.MoodWhenDesireFulfilled.Sum(x => x) /
```

Comment:
This variable name... proud of it? ;-)

Reply:
Please explain this calculation, I'm sure it is right, but I would appreciate understanding it a bit more.

Reply:
I'm calcuating the average mood the viking had when a desire is fulfilled. I'm doing this by adding together all values and dividing the sum with the amount of values.